kyo-prelude/shared/src/main/scala/kyo/Emit.scala

```diff
@@ -185,6 +120,33 @@ object Emit:
 
     inline def runAck[V >: Nothing]: RunAckOps[V] = RunAckOps(())
 
+    final class RunFirstOps[V](dummy: Unit) extends AnyVal:
+
+        /** Runs an Emit effect, capturing only the first emitted value and returning a continuation.
+          *
+          * @param v
+          *   The computation with Emit effect
+          * @return
+          *   A tuple containing:
+          *   - Maybe[V]: The first emitted value if any (None if no values were emitted)
+          *   - A continuation function that takes an Ack and returns the remaining computation
+          */
+        def apply[A: Flat, S](v: A < (Emit[V] & S))(using tag: Tag[Emit[V]], frame: Frame): (Maybe[V], Ack => A < (Emit[V] & S)) < S =
+            ArrowEffect.handleFirst(tag, v)(
+                handle = [C] =>
+                    (input, cont) =>
+                        // Effect found, return the input an continuation
+                        (Maybe(input), cont),
+                done = r =>
+                    // Effect not found, return empty input and a placeholder continuation
+                    // that returns the result of the computation
+                    (Maybe.empty[V], (_: Ack) => r: A < (Emit[V] & S))
+            )
+        end apply
+    end RunFirstOps
+
+    inline def runFirst[V >: Nothing]: RunFirstOps[V] = RunFirstOps(())
```

Comment:
`runHead`?

Reply:
I'd prefer to keep a consistent naming with `ArrowEffect.handleFirst` and `Emit.runFirst`. The "first" is more about the first suspension than the first element.